crates/bevy_winit/src/system.rs

```diff
@@ -164,12 +164,52 @@ pub(crate) fn changed_window(
                     winit_window.set_fullscreen(new_mode);
                 }
             }
+
             if window.resolution != cache.window.resolution {
+                // If `scale_factor_override` changed, the constraints need to be updated with the size.
+                if window.resolution.scale_factor_override()
+                    != cache.window.resolution.scale_factor_override()
+                {
+                    let (min_inner_size, max_inner_size) = get_inner_size_constraints(&window);
+
+                    winit_window.set_min_inner_size(Some(min_inner_size));
+                    winit_window.set_max_inner_size(max_inner_size);
+                }
+
+                // Winit doesn't check constraints when using `set_inner_size`.
+                // Check them manually.
+                let constraints = window.resize_constraints.check_constraints();
+                let logical_width = if window.resolution.width() > constraints.max_width {
```

Comment:
These conditionals could be simplified with `clamp` I believe.

Reply:
Fixed in https://github.com/bevyengine/bevy/pull/8948/commits/ae42333a93f393ac1c2b9114169b122710a706bd

Good catch, I always forget this function exists. ^^'